controller/k8s/api_test.go

```diff
@@ -584,8 +878,20 @@ status:
 				t.Fatalf("api.GetPodsFor() unexpected error, expected [%s] got: [%s]", exp.err, err)
 			}
 
-			if !reflect.DeepEqual(pods, k8sResultPods) {
-				t.Fatalf("Expected: %+v, Got: %+v", k8sResultPods, pods)
+			if len(pods) != len(k8sResultPods) {
+				t.Fatalf("Wrong number of pods: Expected '%d', got '%d'", len(pods), len(k8sResultPods))
+			} else {
+				e, g := make(map[string]*corev1.Pod), make(map[string]*corev1.Pod)
```

Comment:
Is this change because the ordering of the slice is non-deterministic?  A simpler solution might be to sort both slices (e.g. by pod name or uid) before comparing the slices with DeepEqual.  What do you think?

Reply:
I thought about it and gave up because of `[]*corev1.Pod` is not sortable so this solution seemed more practical. But, indeed, it's way simpler. I'm pushing a change with it and a `byPod` new type on the `test_helper.go` just to make it simpler, let me know if you think that we should use a different/existing sortable type.